### PR TITLE
Add a package serialization bit to Module for SIL verifier in Package CMO mode.

### PR DESCRIPTION
**Pull request by elsh**
During Package CMO, SIL cloning happens during which
SILOptions::EnableSerializePackage info is lost.

SILVerifier needs this info to determine whether resilience
can be bypassed for decls serialized in a resiliently
built module when Package CMO optimization enabled.

This PR adds SerializePackageEnabled bit to Module format
and uses that in SILVerifier.

Resolves rdar://126157356

### CODE REVIEW

**Comment by elsh** [comment]
@swift-ci smoke test

**Review by nkcsgexi — APPROVED** [review]

File: lib/SIL/Verifier/SILVerifier.cpp [hunk 1]
```diff
@@ -153,12 +153,20 @@ static bool isArchetypeValidInFunction(ArchetypeType *A, const SILFunction *F) {
...
-                     ResilienceExpansion expansion) {
-  return !D->getModuleContext()->getBypassResilience() &&
+                     ResilienceExpansion expansion,
+                     bool isSerialized) {
+  // Check whether the SIL definition referencing D is serialized but
+  // in minimal resilience expansion context (e.g. M was resiliently
+  // built); this means package serialization was enabled and direct
+  // access should be allowed.
+  auto serializedInResilientMode =
+      expansion == ResilienceExpansion::Minimal && isSerialized;
```
Comment (aschwaighofer) [inline lib/SIL/Verifier/SILVerifier.cpp:167]: The resilience expansion of a ‘serialized’ function is always minimal. (See getResilienceExpansion() in SILFunction.h)

by adding this condition (‘isSerialized()‘) we will by pass the check even in regular resilient mode (no package cmo enabled). I don’t think this is desirable.

 Can you restrict this bypass to only when the current module is in the same package?

**Comment by elsh** [comment]
@swift-ci smoke test

File: lib/SIL/Verifier/SILVerifier.cpp (hunk 1 above)
Comment (elsh) [reply lib/SIL/Verifier/SILVerifier.cpp]: Updated to check for that as well as whether module was built resiliently when serialized.

**Comment by elsh** [comment]
@swift-ci smoke test

File: lib/SIL/Verifier/SILVerifier.cpp (hunk 1 above)
Comment (elsh) [reply lib/SIL/Verifier/SILVerifier.cpp]: Had to add SerializePackageEnabled bit to Module; the SILOpts.EnableSerializePackage info gets lost during SIL cloning stage of Package CMO, so it needed to be serialized to the binary module. It's added to this SILVerifier check function along with others we discussed.

**Comment by elsh** [comment]
@swift-ci smoke test

**Review by aschwaighofer — COMMENTED** [review]
LGTM. Thank you

**Pull request merged**